File: HealthTrac/HealthTrac/Services/UserGoalService.cs
using HealthTrac.DataAccess;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace HealthTrac.Services
{
    public class UserGoalService : IUserGoalService
    {

        private IUserGoalAccessor _acc;

        public UserGoalService(IUserGoalAccessor acc)
        {
            _acc = acc;
        }
        public Models.UserGoal GetUserGoal(long id)
        {
            return _acc.GetUserGoal(id);
        }

        public IEnumerable<Models.UserGoal> GetUserGoals()
        {
            return _acc.GetUserGoals();
        }

        public IEnumerable<Models.UserGoal> GetUserGoals(long goalId)
        {
            return _acc.GetUserGoals(goalId);
        }

        public IEnumerable<Models.UserGoal> GetUserGoals(string userId)
        {
            return _acc.GetUserGoals(userId);
        }

        public Models.UserGoal GetUserGoal(long goalId, string userId)
        {
            return _acc.GetUserGoal(goalId, userId);
        }

        public Models.UserGoal CreateUserGoal(Models.UserGoal ub)
        {
            return _acc.CreateUserGoal(ub);
        }

        public Models.UserGoal UpdateUserGoal(Models.UserGoal ub)
        {
            return _acc.UpdateUserGoal(ub);
        }
    }
}

File: HealthTrac_Mobile/app/src/main/java/com/raik383h_group_6/healthtracmobile/service/api/UserGoalService.java
package com.raik383h_group_6.healthtracmobile.service.api;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.raik383h_group_6.healthtracmobile.model.UserGoal;

import java.util.List;

import retrofit.client.Response;
import retrofit.converter.Converter;
import retrofit.converter.GsonConverter;
import retrofit.http.Body;
import retrofit.http.GET;
import retrofit.http.Header;
import retrofit.http.POST;
import retrofit.http.PUT;
import retrofit.http.Path;
import retrofit.http.Query;

import static com.google.gson.FieldNamingPolicy.UPPER_CAMEL_CASE;

public interface UserGoalService {
    Gson GSON = new GsonBuilder()
            .setFieldNamingPolicy(UPPER_CAMEL_CASE).setDateFormat(ApiConstants.ASPNET_DATE_FORMAT)
            .create();
    Converter DATA_CONVERTER = new GsonConverter(GSON);

    String SERVICE_ENDPOINT = "https://se6.azurewebsites.net";

    @GET("/api/UserGoals/{id}")
    UserGoal getUserGoal(@Path("id") long id, @Header("Authorization") String token);

    @GET("/api/UserGoals")
    UserGoal getUserGoal(@Query("userId") String userId, @Query("goalId") long goalId, @Header("Authorization") String token);

    @GET("/api/UserGoals")
    List<UserGoal> getUserGoals(@Header("Authorization") String token);

    @GET("/api/UserGoals")
    List<UserGoal> getUserGoals(@Query("userId") String userId, @Header("Authorization") String token);

    @GET("/api/UserGoals")
    List<UserGoal> getUserGoals(@Query("goalId") long goalId, @Header("Authorization") String token);

    @PUT("/api/UserGoals/{id}")
    Response updateUserGoal(@Path("id") long id, @Body UserGoal userGoal, @Header("Authorization") String token);

    @POST("/api/UserGoals")
    UserGoal createUserGoal(@Body UserGoal UserGoal, @Header("Authorization") String token);

}
